### PR TITLE
Add health check test and update module direct method "healthcheck" to add correlation Id

### PR DESCRIPTION
**Pull request by HadwaAbdelhalem**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: healthCheck.ps1 [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+#Start the sender process
+$senderProcess = New-Object System.Diagnostics.Process
+$senderProcess.StartInfo = $senderStartInfo
+
+Write-Host "Test commnunication from IOT Hub:$iotHubName to module: $moduleId on device:$deviceId by invoke module direct method:$methodName ..."
+
+$senderProcess.Start() | Out-Null
+$senderOutput = $senderProcess.StandardOutput.ReadToEnd()
+$senderProcess.WaitForExit($senderWaitValue)
```
Comment (blackchoey) [inline healthCheck.ps1]: Can we wait for process exit without setting the timeout value? It should help avoid random failures caused by long invoke-module-method execution time.

File: healthCheck.ps1 [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+$senderStartInfo.CreateNoWindow = $true
+
+#Start the sender process
+$senderProcess = New-Object System.Diagnostics.Process
+$senderProcess.StartInfo = $senderStartInfo
+
+Write-Host "Test commnunication from IOT Hub:$iotHubName to module: $moduleId on device:$deviceId by invoke module direct method:$methodName ..."
+
+$senderProcess.Start() | Out-Null
+$senderOutput = $senderProcess.StandardOutput.ReadToEnd()
```
Comment (blackchoey) [inline healthCheck.ps1]: Do we need to wait for exit first before reading the output so we can get the full output message?

File: healthCheck.ps1 [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+ {
+    Write-Host "Module direct Method Invoked - Hub to Device connection validation failed:"+$senderOutput -ForegroundColor Red
+    exit 1
+
+ }
+
+$senderProcess.close()
+$senderProcess.Dispose()
+
+$listenerOutput = $process.StandardOutput.ReadToEnd()
```
Comment (blackchoey) [inline healthCheck.ps1:109]: Same here, wait for exit first to gain full output message?

File: healthCheck.ps1 [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+$process.WaitForExit($listenerWaitValue)
```
Comment (blackchoey) [inline healthCheck.ps1]: Is the 2000ms timeout enough for the message to arrive? The major consideration here is: if something caused high latency for the messages, we don't expect the health check to fail. Maybe we could try several times before fail the health check. Technically the first release pipeline execution after sample project created should always success.

File: healthCheck.ps1 (hunk 1 above)
Comment (HadwaAbdelhalem) [reply healthCheck.ps1]: fixed

File: healthCheck.ps1 (hunk 2 above)
Comment (HadwaAbdelhalem) [reply healthCheck.ps1]: fixed

File: healthCheck.ps1 (hunk 3 above)
Comment (HadwaAbdelhalem) [reply healthCheck.ps1:109]: fixed

File: healthCheck.ps1 (hunk 4 above)
Comment (HadwaAbdelhalem) [reply healthCheck.ps1]: I added a param to configure the wait time for the listener to receive messages from the IOThub. the  that will ensure the listener waits the max needed time to verify before failing the test if the message was not received. that should cover that issue

File: healthCheck.ps1 [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+$az_iot_ext_install_status=$(az extension show --name azure-cli-iot-ext)
+if ($az_iot_ext_install_status.Length -eq 0)
+    {
+        write-host "Add iot-extension for AZ CLI"
+        az extension add --name azure-cli-iot-ext
+    }
+
+
+# Listener wait duration
+$listenerWaitValue = 2000
```
Comment (blackchoey) [inline healthCheck.ps1:42]: Do we also need to set this to maxTestWaitTimeInSeconds?

Comment (HadwaAbdelhalem) [reply healthCheck.ps1:42]: no, The maxTestWaitTimeInSeconds sets the time the monitor-event commands continue execution.while this wait time is for the process itself  to  wait after the command has completed execution.

Comment (blackchoey) [reply healthCheck.ps1:42]: Thanks for explanation.

**Pull request merged**